### ISSUE TITLE
[develop] docker-compose fails to build webapp resulting in non functional UI.

### ISSUE THREAD

**Issue by r8tor** [owtf/owtf#935]
## Expected Behavior

The system should build and run properly whenever  a new user clone the git repository and run the system using docker-composer up as described in the documentation.

## Current Behavior

When a user clone the git repository and run docker-compose up, the application seems to build properly, however it fails to load the UI when users try to access http://localhost:8009/ with the error bellow:

`Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.5.2-py2.7-linux-x86_64.egg/tornado/web.py", line 1509, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/owtf/owtf/api/handlers/ui_handlers.py", line 18, in get
    self.render('index.html')
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.5.2-py2.7-linux-x86_64.egg/tornado/web.py", line 724, in render
    html = self.render_string(template_name, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.5.2-py2.7-linux-x86_64.egg/tornado/web.py", line 862, in render_string
    t = loader.load(template_name)
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.5.2-py2.7-linux-x86_64.egg/tornado/template.py", line 427, in load
    self.templates[name] = self._create_template(name)
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.5.2-py2.7-linux-x86_64.egg/tornado/template.py", line 454, in _create_template
    with open(path, "rb") as f:
IOError: [Errno 2] No such file or directory: '/owtf/owtf/webapp/build/index.html'`

## Possible Solution

The error indicates a missing file : wtf/owtf/webapp/build/index.html'
Some superficial investigation shows that the yarn build that is supposed to generate that file is failing.  The yarn error logs are available at:  https://github.com/owtf/owtf/blob/develop/owtf/webapp/yarn-error.log 

I tried to trace the issue and it seems like the root problems is on UglifyJS, but I couldn't find much more than that.

## Steps to Reproduce (for bugs)

Run the following commands(tested on OSX)
1. git clone https://github.com/owtf/owtf -b develop
2.cd owtf
3. docker-composer up
4. Launch browser and navigate to http://localhost:8009/

## Context

Unable to build develop branch

## Your Environment

* Installation method used: Docker Compose
* Version/branch used: develop
* Operating System and version (like Kali, Debian, ArchLinux, etc): Kali container on OSX(docker community, version 17.12.0-ce-mac55 (23011))
* pip/setuptools version: latest on kali container
* sudo access: container user is is root.

**Comment by viyatb** [comment]
@r8tor thanks for the report. The UI in the develop branch should load but is broken since it is currently being ported to React (from the Tornado templates).

While the UI remains broken, the `index.html` file should be correctly generated by webpack. I'll look into it.

**Comment by r8tor** [comment]
Hey Viyat, 

Thanks for looking into to it and for maintaining such amazing software :-)

I hope I’ll have the chance to buy you a coffee(or tea) or a beer in a conference someday.

Cheers, 
RT

**Comment by fristonio** [comment]
@viyatb Can I try to fix this issue?

**Comment by viyatb** [comment]
Sure, go ahead!  Thanks.

**Comment by saganshul** [comment]
@viyatb @fristonio There is not the only problem with docker configuration. There are a lot of problems with the installation script.  Look at this `ui_setup()` function at https://github.com/owtf/owtf/blob/develop/scripts/install.sh#L332 . Also to build for production we need to use `yarn run prod`. But it will not work. Because of command for `prod` contains `-p`
`"prod": "webpack --progress -p --config webpack/common.config.js",` which denotes
`-p : --optimize-minimize --define process.env.NODE_ENV="production"` and in `prod.config.js` we already have
```
new webpack.optimize.UglifyJsPlugin({
            compress: {
                warnings: false
            }
        })
```
So, Basically copying of code. That's why it `yarn run prod` fails. 
@fristonio This is also an answer for your #937. Hope it clarifies everything. I will submit a PR to fix all this in few hours.

**Comment by fristonio** [comment]
I had this figured out and was about to make a PR :relieved: but its good that you did it. There are some more issues with the installation script as I discussed with viytab on slack. Also docker-compose.yml is not correct as it maps port 8080 to 8080 but owtf uses 8008 port. I will have it fixed and will make a PR. :)

**Comment by viyatb** [comment]
@fristonio ah, I just noticed the `8008 -> 8080` typo. Send the PR ;)!

**Comment by fristonio** [comment]
@viyatb I think we can close this now. :)

**Comment by viyatb** [comment]
Closed in fc08d08e0b710b715d39e4044309f8a2ac4b95dc.
